cmake/Platform/Extras/CheckPathExistsCaseSensitive.cmake

```diff
@@ -0,0 +1,132 @@
+#=============================================================================#
+# _check_path_exists_case_sensitive_brute_force
+# [PRIVATE/INTERNAL]
+#
+# _check_path_exists_case_sensitive_brute_force(result_var_ absolute_path_)
```

Comment:
It seems like you forgot to answer me in my last review lol - Why are you naming parameters ending with an underscore? Where does this convention come from?

Reply:
Also, most of **Arduino-CMake**'s functions "accept" the `result_var/output_var` parameter as the last one - Could you match the project's convention here as well?

Reply:
In all CMake related projects I am working on I add a trailing underscore to local function variables to distinguish them from variables in parent scope or cache variables. Only for cache variables I use upper case. Although CMake is not case-sensitive when it comes to variable and function names, this proved to be useful in many occasions. But of course, I can go without this and modify this PR.

## Another thing about variable names
In most parts of Arduino-CMake, variables names are uppercase. Using uppercase variables and function names in CMake is a reminiscence of the old days where CMake required it. Most CMake code that is witten these days is in all lowercase, which is much more convenient to read. I converted most of my old projects to all lowercase using [this simple method](https://stackoverflow.com/questions/22839458/tools-for-upper-lower-case-consistency-in-cmake-source).

## About where to put the result variable
I wasn't aware that there is something like a convention in Arduino-CMake.

I haven't conducted a thorough statistical investigation, but it seems to me that CMake by itself puts the result variable first in most of its functions. This is, however, not done consistently because there are are some functions in CMake that are quite old and from a time where the developers did work with heterogenous standards.

The reason why I prefer the return value first, is that it follows C/C++(pre C++11)/Java/etc. style where the return value is the first thing that appears in a function definition.

Having a convention here helps in any case. Very often I find myself trying to remember the order of parameters of a CMake function that I used a thousand times before. This is because of the slight inconsistencies in CMakes I mentioned above. As CMake is a non-strictly typed language, accidentally swapping arguments can go undetected and lead to very nasty errors. That's why I prefer to use [cmake_parse_arguments](https://cmake.org/cmake/help/v3.0/module/CMakeParseArguments.html) in cases where I have many arguments. Using named arguments makes calling much safer.

Reply:
Great info there, thank you!
Will you be able to convert all of our source to lower-case some when? I would do it myself but currently I don't have any Linux at hand 😢 

As for the variable positioning:
I agree with you that it's better placed at the beginning of the parameters to note that it's a return value, though I must disagree with you on the statistical subject as the examples you gave in your comment are one of the few that don't line with my statement.
However, none of this matters at I'll modify all functions to "accept" the return variable as the 1st one, probably after converting the source to lower-case, and also update the `CONTRIBUTING.md` file to note this stuff.

Reply:
> Will you be able to convert all of our source to lower-case some when? I would do it myself but currently I don't have any Linux at hand 😢

Yes I can do that. There is however one problem. Any stuff that was branched before the transition will need manual work (passing single files through the automatic change scripts).
But I think that is a risk we can take.